plugin/mutate/source/dextool/plugin/mutate/backend/database/standalone.d

```diff
@@ -535,22 +547,21 @@ struct Database {
     }
 
     /** Set mutantStatus by using MutantId.
-	Can be used both for ignoring and "accepting" mutants.
      */
-    void markMutant(ulong id, Mutation.Status to_status, string rationale = "") @trusted {
-        Mutation.Status from_status = Mutation.Status(getMutationStatus(MutationId(id)));
+    void markMutant(MutationId id, Mutation.Status to_status, string rationale) @trusted {
+        auto st_id = getMutationStatusId(id);
+        auto p = getPath(id);
 
-        // mark mutantStatus in mutationStatus-table
-        const s = format!"UPDATE %s SET status=%s WHERE id IN (SELECT st_id FROM %s WHERE id = %s)"(
-                mutationStatusTable, to_status.to!long, mutationTable, id.to!long);
-        auto stmt = db.prepare(s);
-        stmt.execute;
+        if (p.isNull)
+            return;
+
+        auto sl = getSourceLocation(id);
+        auto r = Rationale(rationale);
+        auto t = Clock.currTime.toUTC;
 
-        // add mutant to separate table for marked mutants
-        db.run(buildSchema!MarkedMutant); // TODO: can probably be done somewhere else
-        db.run(insert!MarkedMutant.insert, MarkedMutant(MutationId(id),
-                Path(getPath(MutationId(id))), from_status, to_status,
-                Rationale(Clock.currTime.toUTC.toSqliteDateTime, rationale)));
+        // update mutationStatus and add mutant to separate table for marked ones.
+        updateMutationStatus(st_id, to_status);
```

Comment:
I recommend moving this logic to the caller. Let it decide what to do.
It reduces the encoded business logic in the database and allow a higher degree of flexibility of the caller.
This method might, for example, be used in currently unknown ways where we do not want to changes the status :)

Reply:
Hmm, I am not sure I completely understand. I pushed more commits, so I just commented it out for now. But do we mean removing it?

Reply:
Ah, you mean move it to the markMutant-function in ../admin.d ? (as an example)